databuilder/extractor/neo4j_dashboard_search_data_extractor.py

```diff
@@ -0,0 +1,88 @@
+import textwrap
+from typing import Any  # noqa: F401
+
+from pyhocon import ConfigTree  # noqa: F401
+
+from databuilder import Scoped
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.neo4j_extractor import Neo4jExtractor
+from databuilder.publisher.neo4j_csv_publisher import JOB_PUBLISH_TAG
+
+
+class Neo4jDashboardSearchDataExtractor(Extractor):
+    """
+    Extractor to fetch data required to support search from Neo4j graph database
+    Use Neo4jExtractor extractor class
+    """
+    CYPHER_QUERY_CONFIG_KEY = 'cypher_query'
+
+    DEFAULT_NEO4J_CYPHER_QUERY = textwrap.dedent(
```

Comment:
Hi @feng-tao, sorry for late answers, it was a busy week. 

1. metric <-> dashboard relation is n:1 (dashboard can have more metrics). dashboard <-> dashboardgroup is 1:1 (one dashboard can have one group) - in this case it could 1:n, dashboards can be assigned to more groups (but not in Qlik Sense case).
2. I named it dashboardgroup to have it more generic, but yes, Qlik Sense has something what they call `stream`. Simply group of dashboards. 1:1 relation is explained [here](https://help.qlik.com/en-US/sense/June2019/Subsystems/ManagementConsole/Content/Sense_QMC/streams-overview.htm) 2nd paragraph `An app can be published to only one stream`.
3. In Qlik Sense they have really nice APIs to get/set almost everything, there is an [example](https://gist.github.com/jhettler/346799f622098ed65e727f203eabbd02) how to load the dashboard data  - this is Qlik QRS (Qlik Repository Service) API [here](https://help.qlik.com/en-US/sense-developer/June2019/Subsystems/RepositoryServiceAPI/Content/Sense_RepositoryServiceAPI/RepositoryServiceAPI-Introduction.htm) if you look for `QRS API reference` you will find all API methods and we use qrspy as a wrapper in python [here](https://github.com/clintcarr/qrspy/blob/master/qrspy.py). For metrics it is a bit more complicated, because you have to get inside Qlik Sense application (it is dashboard, but Qlik call it application), to get the information about metrics we use `Qlik Engine JSON API` [here](https://help.qlik.com/en-US/sense-developer/June2019/Subsystems/EngineAPI/Content/Sense_EngineAPI/introducing-engine-API.htm) and [this](https://github.com/qliknln/pyqlikengine) to call it from python - we plan to take the further maintenance. But unfortunatelly, as I know, they have no API how to access usage statistics. They have something called `Monitoring apps` and in this stream, they have `Operations monitor` as a dashboard where you can find all the statistics, but it is parsed from Qlik service logs (files). So you would have to probably write own parser and create new API. But we have never asked Qlik if they could provide the parsing examples etc. as we had no needs to use it.